docs/devel/e2e.md

```diff
@@ -32,33 +32,92 @@ Documentation for other releases can be found at
+The e2e tests in kubernetes are built atop of [Ginkgo](http://onsi.github.io/ginkgo/) and [Gomega](http://onsi.github.io/gomega/).  There are a host of features that this BDD testing framework provides, and it is recommended that the developer read the documentation prior to diving into the tests.
 
-The purpose of *this* document is to serve as a primer for developers who are looking to execute, or add tests, using a local development environment.
+The purpose of *this* document is to serve as a primer for developers who are looking to execute or add tests using a local development environment.
 
 ## Building and Running the Tests
 
-**NOTE:** The tests have an array of options.  For simplicity, the examples will focus on leveraging the tests on a local cluster using `sudo ./hack/local-up-cluster.sh`
+There are a variety of ways to run e2e tests, but we aim to decrease the number of ways to run e2e tests to a canonical way: `hack/e2e.go`.
 
-### Building the Tests
+You can run an end-to-end test which will bring up a master and nodes, perform some tests, and then tear everything down. Make sure you have followed the getting started steps for your chosen cloud platform (which might involve changing the `KUBERNETES_PROVIDER` environment variable to something other than "gce").
 
-The tests are built into a single binary which can be run against any deployed kubernetes system.  To build the tests, navigate to your source directory and execute:
+To build Kubernetes, up a cluster, run tests, and tear everything down, use:
 
-`$ make all`
+```sh
+go run hack/e2e.go -v --build --up --test --down
+```
+
+If you'd like to just perform one of these steps, here are some examples:
+
+```sh
+# Build binaries for testing
+go run hack/e2e.go --build
+
+# Create a fresh cluster.  Deletes a cluster first, if it exists
+go run hack/e2e.go --up
+
+# Create a fresh cluster at a specific release version.
+go run hack/e2e.go --up --version=0.7.0
+
+# Test if a cluster is up.
+go run hack/e2e.go --isup
+
+# Push code to an existing cluster
+go run hack/e2e.go --push
+
+# Push to an existing cluster, or bring up a cluster if it's down.
+go run hack/e2e.go --pushup
+
+# Run all tests
+go run hack/e2e.go --test
 
-The output for the end-2-end tests will be a single binary called `e2e.test` under the default output directory, which is typically `_output/local/bin/linux/amd64/`.  Within the repository there are scripts that are provided under the `./hack` directory that are helpful for automation, but may not apply for a local development purposes.  Instead, we recommend familiarizing yourself with the executable options.  To obtain the full list of options, run the following:
+# Run tests matching the regex "\[Conformance\]" (the conformance tests)
+go run hack/e2e.go -v -test --test_args="--ginkgo.focus=\[Conformance\]"
```

Comment:
ug.. we need to manually escape vs. going to the tests themselves?  

Why would we be pointing them to hack/e2e.go vs. just pointing them to ginkgo/e2e.test directly?  I really don't see why this is easier, or better...?